garglk/sysgtk.c

```diff
@@ -608,12 +611,6 @@ void winopen(void)
 
     wintitle();
 
-    gtk_window_set_geometry_hints(GTK_WINDOW(frame),
```

Comment:
The only regression I'm seeing in KDE is that if the window is too large (i.e. cols and/or rows cause it to overflow the screen), the window manager doesn't/can't resize it to fit.  I don't think one way is _better_ than the other, but I'd prefer to keep the previous behavior for continuity, if possible.

I did some small tests and it's easy to get it back working in KDE.  Include `gdk/gdkwayland.h` and reinstate this deleted code, protected by `if gtk_window_set_geometry_hints(GTK_WINDOW(frame)`; i.e. run this only under non-Wayland.

Then, above, call `gtk_widget_set_size_request(canvas, defw, defh)` only if it _is_ Wayland.  The rest of the changes are fine in KDE.

If that works under Wayland with Gtk then we should be good to merge.